rs/src/parser.rs

```diff
@@ -102,6 +102,20 @@ fn define_lambda(i: &str) -> IResult<&str, Expr> {
     Ok((i, Expr::Lambda(Code { name, tail: false, formals, body, free: vec![] })))
 }
 
+fn define_variadic_fn(i: &str) -> IResult<&str, Expr> {
+    let (i, _) = tuple((open, tag("define"), space1))(i)?;
+    let (i, mut params) = delimited(open, identifiers, tag("."))(i)?;
+    let (i, rest_param) = delimited(space0, identifier, close)(i)?;
+    let (i, body) = delimited(space0, many1(terminated(expression, space0)), space0)(i)?;
```

Comment:
You might need `space1` instead of `space0` in `let (i, rest_param) = delimited(space0, identifier, close)(i)?;` to make sure that there is atleast one space after the `.`. 

I havent tested this yet, but could you try what happens for `(define (add x y .args) (reduce + 0 args))` please?

Reply:
![image](https://user-images.githubusercontent.com/601714/81673993-6ae06800-9444-11ea-9371-8f8ec87734f4.png)

The space is necessary, otherwise `.args` becomes one identifier instead of what we want here.